app/movie/[imdbid]/page.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import * as React from "react";
+import { MovieField } from "@/lib/MovieField";
+import {HiMiniHeart,HiOutlineHeart,HiMiniStar,HiOutlineStar } from "react-icons/hi2";
+import { MoviesContext } from "@/context/movieData.context";
+import movieData from "../../../lib/top_100_movies.json";
+
+interface PropType {
+  params: {
+    imdbid: string;
+  };
+}
+
+const MovieDetail: React.FC<PropType> = ({ params }) => {
+  const [isFavourited, setIsFavourited] = React.useState(false);
+  const {addFavMovie, removeFavMovie} = React.useContext(MoviesContext);
+  const movie = movieData.find((movie) => movie.imdbid === params.imdbid);
+  const {
+    title,
+    rating,
+    id,
+    year,
+    image,
+    description,
+    trailer,
+    genre,
+    director,
+    writers,
+    imdbid
+  } = movie as MovieField;
+
+  const handleAddFav = (_movie: MovieField) => {
```

Comment:
You've added the Favorite logic here but the PR description only mentions the search/filter. Shall I review the favorite logic too or is it still being done? I noticed this isn't working yet.

Reply:
still working and I think I will do all of them for this branch